### PR TITLE
Feature/#166 bootstrap forms

### PR DESCRIPTION
**Pull request by Harikarthyk**

### CODE REVIEW

File: application/backend/src/main/java/io/typerefinery/websight/models/components/BaseComponent.java [hunk 1]
```diff
@@ -137,6 +130,23 @@ public String getComponentClassNames() {
         return "";
     }
 
+    private Map<String, String> gridConfig = new HashMap<String, String>() {
```
Comment (wildone) [inline application/backend/src/main/java/io/typerefinery/websight/models/components/BaseComponent.java:133]: can this be in grid class?

File: application/backend/src/main/java/io/typerefinery/websight/models/components/BaseComponent.java [hunk 2]
```diff
@@ -137,6 +130,23 @@ public String getComponentClassNames() {
...
+        {
+            put("lgColSize", "col-lg-");
+            put("mdColSize", "col-md-");
+            put("smColSize", "col-sm-");
+        }
+    };
+
+    private Map<String, String> textAlignmentConfig = new HashMap<String, String>() {
```
Comment (wildone) [inline application/backend/src/main/java/io/typerefinery/websight/models/components/BaseComponent.java:141]: can this be in style class?

File: application/backend/src/main/java/io/typerefinery/websight/models/components/BaseComponent.java [hunk 3]
```diff
@@ -153,7 +163,17 @@ protected void init() {
             componentClasses = Arrays.stream(style.getClasses())
             .collect(Collectors.toCollection(LinkedHashSet::new))
             .toArray(new String[]{});
-        }
+
+            // Width
+            grid.addClasses(gridConfig.get("lgColSize") + getLgColSize());
```
Comment (wildone) [inline application/backend/src/main/java/io/typerefinery/websight/models/components/BaseComponent.java:168]: can this be done in grid class?

File: application/backend/src/main/java/io/typerefinery/websight/models/components/BaseComponent.java [hunk 4]
```diff
@@ -163,9 +183,10 @@ protected void init() {
         }
 
         // add default class name as first class
```
Comment (wildone) [inline application/backend/src/main/java/io/typerefinery/websight/models/components/BaseComponent.java:185]: is this not needed?

File: application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Button.java [hunk 5]
```diff
@@ -18,136 +18,98 @@
...
-import org.apache.sling.models.annotations.Default;
-import org.apache.sling.models.annotations.Model;
-
-import io.typerefinery.websight.models.components.BaseFormComponent;
-import io.typerefinery.websight.utils.GridDisplayType;
-import io.typerefinery.websight.utils.GridStyle;
+import org.apache.sling.api.SlingHttpServletRequest;
 
-@Model(adaptables = Resource.class, defaultInjectionStrategy = OPTIONAL)
+@Model(adaptables = { Resource.class, SlingHttpServletRequest.class }, defaultInjectionStrategy = OPTIONAL)
```
Comment (wildone) [inline application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Button.java:39]: can this be formatted please.

File: application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Form.java [hunk 6]
```diff
@@ -17,13 +17,18 @@
...
 import org.apache.sling.api.resource.Resource;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.ExporterOption;
 import org.apache.sling.models.annotations.Model;
 import javax.annotation.PostConstruct;
 
 import io.typerefinery.websight.models.components.BaseFormComponent;
 
-@Model(adaptables = Resource.class, defaultInjectionStrategy = OPTIONAL)
+@Model(adaptables = {Resource.class, SlingHttpServletRequest.class}, defaultInjectionStrategy = OPTIONAL)
```
Comment (wildone) [inline application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Form.java:30]: can this be formatted?

File: application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Form.java [hunk 7]
```diff
@@ -60,5 +65,8 @@ protected void init() {
       this.id = DEFAULT_ID;
       this.module = DEFAULT_MODULE;
       super.init();
+      if(grid != null) {
+        grid.addClasses("card p-4 mb-3");
```
Comment (wildone) [inline application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Form.java:69]: why is this?

File: application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Input.java [hunk 8]
```diff
@@ -18,52 +18,30 @@
...
 import org.apache.sling.models.annotations.Exporter;
 import org.apache.sling.models.annotations.ExporterOption;
-import org.apache.sling.models.annotations.Model;
 
 import org.apache.sling.api.SlingHttpServletRequest;
 
-import org.jetbrains.annotations.Nullable;
-import javax.inject.Named;
-
-@Model(
```
Comment (wildone) [inline application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Input.java:49]: formatting?

File: application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Label.java [hunk 9]
```diff
@@ -18,110 +18,40 @@
...
-import org.apache.sling.models.annotations.Model;
 
 import org.apache.sling.api.SlingHttpServletRequest;
-import org.apache.sling.models.annotations.injectorspecific.Self;
-import org.apache.sling.models.annotations.injectorspecific.RequestAttribute;
-
-import org.jetbrains.annotations.Nullable;
-import javax.inject.Named;
 
-@Model(
```
Comment (wildone) [inline application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Label.java:51]: formatting?

**Review by wildone — CHANGES_REQUESTED** [review]

File: application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Form.java (hunk 7 above)
Comment (Harikarthyk) [reply application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Form.java]: The default classes for the Form container.

Can I make this in variable add assign them ?

File: application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Form.java (hunk 6 above)
Comment (Harikarthyk) [reply application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Form.java]: Yeah done.

File: application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Button.java (hunk 5 above)
Comment (Harikarthyk) [reply application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Button.java]: yeah done.

File: application/backend/src/main/java/io/typerefinery/websight/models/components/BaseComponent.java (hunk 1 above)
Comment (Harikarthyk) [reply application/backend/src/main/java/io/typerefinery/websight/models/components/BaseComponent.java]: I have moved the config things to StyledGrid Classes.

File: application/backend/src/main/java/io/typerefinery/websight/models/components/BaseComponent.java (hunk 2 above)
Comment (Harikarthyk) [reply application/backend/src/main/java/io/typerefinery/websight/models/components/BaseComponent.java]: I have moved the all config things to StyledGrid Classes.

File: application/backend/src/main/java/io/typerefinery/websight/models/components/BaseComponent.java (hunk 3 above)
Comment (Harikarthyk) [reply application/backend/src/main/java/io/typerefinery/websight/models/components/BaseComponent.java]: now all these things are calculated from the functions which is defined in the DefaultStyledGridComponent

File: application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Form.java (hunk 7 above)
Comment (Harikarthyk) [reply application/backend/src/main/java/io/typerefinery/websight/models/components/forms/Form.java]: I have created DEFAULT_FORM_CLASSES and assigned all the default classes for the form.

**Review by wildone — APPROVED** [review]

**Pull request merged**